File: 1-TwoSum/1-TwoSum.java
// Last updated: 6/4/2026, 7:25:38 PM
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<nums.length; i++){
            int needed = target - nums[i];
            if(map.containsKey(needed)){
                return new int[]{i, map.get(needed)};
            }
            map.put(nums[i],i);
        }
        return new int[0];
    }
}

File: 1-TwoSum/1-TwoSum.c
// Last updated: 6/4/2026, 7:25:41 PM
#include <stdlib.h>

int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    int* answer = (int*)malloc(2 * sizeof(int));
    *returnSize = 2;
    for (int i = 0; i < numsSize; i++) {
        for (int j = i + 1; j < numsSize; j++) {
            if (nums[i] + nums[j] == target) {
                answer[0] = i;
                answer[1] = j;
                return answer;
            }
        }
    }
    return NULL;
}

